### PR TITLE
feat(swap): load existing orders for quoter

### PR DESCRIPTION
**Pull request by acdibble**

### CODE REVIEW

File: packages/swap/src/quoting/Leg.ts [hunk 1]
```diff
@@ -32,6 +32,14 @@ export default class Leg {
     return this.destAsset !== 'Usdc' ? 'BUY' : 'SELL';
   }
 
+  getBaseAsset(): Exclude<InternalAsset, 'Usdc'> {
+    if (this.destAsset !== 'Usdc') return this.destAsset;
+
+    if (this.srcAsset !== 'Usdc') return this.srcAsset;
+
+    throw new Error('one of the assets must be Usdc');
```
Comment (niklasnatter) [inline packages/swap/src/quoting/Leg.ts:40]:
```suggestion
    throw new Error('one of the assets must not be Usdc');
```

File: packages/swap/src/utils/rpc.ts [hunk 2]
```diff
@@ -39,3 +40,20 @@ export const getEgressFee = async (asset: InternalAsset): Promise<bigint | null>
 
   return readChainAssetValue(environment.ingressEgress.egressFees, asset);
 };
+
+const fetchOrdersAndPrice = async (baseAsset: Exclude<InternalAsset, 'Usdc'>) => {
+  const [orders, price] = await Promise.all([
+    getPoolOrders(rpcConfig, getAssetAndChain(baseAsset), getAssetAndChain('Usdc')),
```
Comment (niklasnatter) [inline packages/swap/src/utils/rpc.ts:46]: maybe we should give them a suffix to indicate that the result is not parsed

Comment (niklasnatter) [reply packages/swap/src/utils/rpc.ts:46]: but no strong opinion as theay are typed anyway i think?

File: packages/swap/src/utils/rpc.ts [hunk 3]
```diff
@@ -39,3 +40,20 @@ export const getEgressFee = async (asset: InternalAsset): Promise<bigint | null>
...
+  return { poolState: orders, rangeOrderPrice: price.rangeOrder };
+};
+
+const ordersCacheMap = new AsyncCacheMap({
+  refresh: false,
+  refreshInterval: 6_000,
+  fetch: fetchOrdersAndPrice,
+});
+
+export const getCachedPoolOrdersAndPrice = ordersCacheMap.get.bind(ordersCacheMap);
```
Comment (niklasnatter) [inline packages/swap/src/utils/rpc.ts:59]: what does this do? 🤔

**Review by niklasnatter — APPROVED** [review]

File: packages/swap/src/quoting/Leg.ts (hunk 1 above)
Comment (acdibble) [reply packages/swap/src/quoting/Leg.ts:40]: One of them must be `Usdc`. We return the one that isn't

**Pull request merged**